tests/test_read.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+import pathlib
+import numpy as np
+
+import src.read as rd
+
+def test_preprocess():
+    """
+    Tests preprocess function by asserting title, 
+    shape of corpus, and correct line reading.
+    """
+    txt_path = "tests/data/test.txt"
+    file = open(txt_path)
+    corpus = rd.preprocess(file)
+    file.close()
+
+    assert np.shape(corpus) == (26,)
+    assert corpus[0] == 'Testing Text File'
+    assert corpus[2] == 'Link to generator repo!'
+    assert corpus[20] == 'Here are some Chinese characters: .'
+    assert corpus[22] == 'The vowels: are , , , , , , .'
+    assert corpus[24] == 'We can also test from mathematical symbols: , , , , , X, %,  ,a, , , +, = ,-.'
```

Comment:
Looks like you have a syntax error in your test file. Review that to make sure it's grammatically correct. E.g., this should probably be "for" not "from".

Reply:
Checked: I read through the test files and checked for grammar, spelling, etc.